app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java

```diff
@@ -200,13 +228,28 @@ public Builder withContext(ScheduleContext context) {
         }
         
         public ScheduleContext build() {
-            // pretty much everything else is optional. I would like healthCode to be required, but it's not:
-            // we use these selection criteria to select subpopulations on sign up.
-            if (now == null) {
-                now = (initialTimeZone == null) ? DateTime.now() : DateTime.now(initialTimeZone);
+            // ScheduleContext is validated to have the correct values:
+            // - requestTimeZone is required (when validated)
+            // - now is advisable, but still not required
+            // - endsOn OR daysAhead needs to be supplied to calculate endsOn from now
+            if (initialTimeZone == null) {
+                initialTimeZone = requestTimeZone;
+            }
+            if (requestTimeZone != null) {
+                if (now == null) {
+                    now = DateTime.now(requestTimeZone);
+                } else {
+                    now = now.withZone(requestTimeZone);    
+                }
```

Comment:
There are a few places in the code where we assume ScheduleContext.getNow() is not null. Under the old code, the builder would always ensure now is not null. Under the new code, we cannot guarantee that to be the case. Do we need to fix this?